docs/design/proposals/late-initialization/late-initialization.md

```diff
@@ -0,0 +1,151 @@
+## Introduction
+
+[K8s convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#late-initialization) describes **Late Initialization** as “When resource fields are set by a system controller after an object is created/updated”
+
+In the context of an ACK service controller, when an AWS resource is created or updated by the controller, the AWS service may set optional fields for a resource to a default value and return them as part of Create/Update output or even as Get output in some cases. 
+
+This **late initialization** of optional fields in the AWS resource are not currently reflected in the desired state of the resource. This causes a difference to be detected between the stored desired state of the resource and the latest state fetched from the AWS service at the start of a reconciliation loop when we call the resource manager's ReadOne method. However, this difference is erroneous since the controller does not need to call the resource manager's Update method in order for the actual state to match the desired state of the resource.
+
+This document will propose the solution for above problem and provide the details of implementing it.
+
+## Requirements
+
+A solution for this problem should :
+
+1. Update the desired state of the K8s custom resource with late initialized fields which are optional and not provided by the K8s user. 
+
+2. Provide configuration for AWS service teams to indicate which operation output returns the late initialized fields. i.e. Get/List/Create/Update output.
+
+    > AWS APIs have different behavior for returning the late initialized to the user. Some services return these fields as part of Create output, while some services will return the late initialized fields as part of  ReadOne output after the resource creation.
+
+3. Wait for asynchronous resource creation to populate late initialized fields.
+
+    > For some services/resources the late initialized fields do not appear immediately.  This happens when there is asynchronous processing involved in resource creation. The proposed solution should allow waiting and retry until the late initialized fields appear in the mentioned API output.
+
+4. Allow the AWS service teams to provide custom code using hooks for handing of late initialized fields.
+
+## Solutions
```

Comment:
Recommend changing this section to `## Proposed Solution` and structuring the section's content like so:

### ACK code-generator

> Describe proposed changes to how the code generator discovers server-side defaulting fields>

> Describe proposed changes to the generator config that allows service teams to inform the code generator when it cannot figure out which fields have a server-side defaulting behaviour>

### ACK runtime

> Describe proposed changes to the main ACK runtime reconciler loop for handling server-side default values for resource fields that are set immediately, including before and after pseudo-code>

> Describe proposed changes to the main ACK runtime reconciler loop for handling server-side default values for resource fields that are set after some delay (asynchronously), including before and after pseudo-code